app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,31 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts.includes(:comments)
+    @user = current_user
+    @posts = @user.posts
```

Comment:
- [ ] To solve the N+1 problem, I will suggest you add the `include` method so that a user will be able to fetch all comments that belong to a post at once. Just the way you implemented it initially. like so:

`@posts = @user.posts.includes(:comments)`